src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -6919,6 +6920,230 @@ def create_stack(self, stack: StackRequest) -> StackResponse:
+                                ValueError,
+                                IllegalOperationError,
+                                NotImplementedError,
+                                AuthorizationException,
+                            ) as e:
+                                raise RuntimeError(
+                                    f"Access to the resource could not be "
+                                    f"verified: {e}"
+                                )
+
+                            resources = connector_resources.resources[0]
+                            if resources.resource_ids:
+                                if len(resources.resource_ids) > 1:
+                                    raise RuntimeError(
+                                        f"Multiple {resource_type} resources are "
+                                        "available for the selected connector. "
+                                        "Please use the a specific resource-id to "
+                                        f"configure a {resource_type} resource."
+                                    )
+                                else:
+                                    resource_id = resources.resource_ids[0]
+
+                            component_update = ComponentUpdate(
+                                connector=service_connector.id,
+                                connector_resource_id=resource_id,
+                            )
+                            self.update_stack_component(
+                                component_id=component.id,
+                                component_update=component_update,
+                            )
+
+                    components_mapping[component_type] = [
+                        component.id,
+                    ]
+
+                # Stack
+                stack_name = full_stack.name
+                while True:
+                    try:
+                        stack_request = StackRequest(
+                            user=full_stack.user_id,
+                            workspace=full_stack.workspace_id,
+                            name=stack_name,
+                            description=full_stack.description,
+                            components=components_mapping,
+                        )
+                        stack_response = self.create_stack(stack_request)
+                        break
+                    except EntityExistsError:
+                        stack_name = (
+                            f"{full_stack.name}-{random_str(4)}".lower()
+                        )
+                        continue
+
+                handler.metadata.update(
+                    stack_response.get_analytics_metadata()
+                )
+            except Exception as e:
+                for service_connector_id in service_connectors_created_ids:
+                    self.delete_service_connector(
+                        service_connector_id=service_connector_id
+                    )
```

Comment:
I think you need to delete stack components first, otherwise you run into a validation error because the service connector is still referenced

Reply:
Oh, that makes sense. I am switching them.